pkg/engine/masterarmresources.go

```diff
@@ -0,0 +1,156 @@
+
+		if provisionJumpbox {
+			jumpboxVM := createJumpboxVirtualMachine(cs)
+			masterResources = append(masterResources, jumpboxVM)
+			jumpboxIsManagedDisks :=
+				cs.Properties.OrchestratorProfile.KubernetesConfig.PrivateJumpboxProvision() &&
+					cs.Properties.OrchestratorProfile.KubernetesConfig.PrivateCluster.JumpboxProfile.StorageProfile == api.ManagedDisks
+			if !jumpboxIsManagedDisks {
+				jumpBoxStorage := createJumpboxStorageAccount()
+				masterResources = append(masterResources, jumpBoxStorage)
+			}
+			jumpboxNSG := createJumpboxNSG()
+			masterResources = append(masterResources, jumpboxNSG)
+			jumpboxNIC := createJumpboxNIC(cs)
+			masterResources = append(masterResources, jumpboxNIC)
+			jumpboxPublicIP := createJumpboxPublicIPAddress()
+			masterResources = append(masterResources, jumpboxPublicIP)
+		}
+
+	}
+
+	if p.MasterProfile.Count > 1 {
+		internalLB := CreateMasterInternalLoadBalancer(cs)
+		masterResources = append(masterResources, internalLB)
+	}
+
+	isKMSEnabled := to.Bool(cs.Properties.OrchestratorProfile.KubernetesConfig.EnableEncryptionWithExternalKms)
+
+	if isKMSEnabled {
+		keyVaultStorageAccount := createKeyVaultStorageAccount()
+		masterResources = append(masterResources, keyVaultStorageAccount)
+		keyVault := CreateKeyVaultVMAS(cs)
+		masterResources = append(masterResources, keyVault)
+	}
+
+	masterVM := CreateVirtualMachine(cs)
+	masterResources = append(masterResources, masterVM)
+
+	useManagedIdentity := cs.Properties.OrchestratorProfile.KubernetesConfig.UseManagedIdentity
+	userAssignedIDEnabled := useManagedIdentity && cs.Properties.OrchestratorProfile.KubernetesConfig.UserAssignedID != ""
+
+	if useManagedIdentity && !userAssignedIDEnabled {
+		vmasRoleAssignment := createVMASRoleAssignment()
+		masterResources = append(masterResources, vmasRoleAssignment)
+	}
+
+	masterCSE := CreateCustomScriptExtension(cs)
+	masterResources = append(masterResources, masterCSE)
+
+	if cs.IsAKSBillingEnabled() {
+		aksBillingExtension := CreateAKSBillingExtension(cs)
+		masterResources = append(masterResources, aksBillingExtension)
+	}
+
+	//TODO: Implement Extensions
+
+	return masterResources
+}
+
+func createKubernetesMasterResourcesVMSS(cs *api.ContainerService) []interface{} {
+	var masterResources []interface{}
+
+	//TODO: Implement CosmosEtcd
+
+	masterNsg := CreateNetworkSecurityGroup(cs)
+	masterResources = append(masterResources, masterNsg)
+
+	if cs.Properties.OrchestratorProfile.RequireRouteTable() {
+		masterResources = append(masterResources, createRouteTable())
+	}
+	if !cs.Properties.MasterProfile.IsCustomVNET() {
+		masterVnet := createVirtualNetworkVMSS(cs)
+		masterResources = append(masterResources, masterVnet)
+	}
+
+	if cs.Properties.MasterProfile.Count > 1 {
+		internalLb := CreateMasterInternalLoadBalancer(cs)
+		masterResources = append(masterResources, internalLb)
+	}
+
+	publicIPAddress := CreatePublicIPAddress()
```

Comment:
ditto the above comment to refactor this to (1) serially compose a set of resource types and (2) append to resource array w/ a one-liner